mne_bids/write.py

```diff
@@ -1249,39 +1310,75 @@ def write_raw_bids(raw, bids_basename, bids_root, events_data=None,
+        subject=subject_id, session=session_id, acquisition=acquisition,
+        suffix='electrodes.tsv', prefix=data_path)
+    unit = "m"  # defaults to meters
     # We only write electrodes.tsv and accompanying coordsystem.json
     # if we have an available DigMontage
     if raw.info['dig'] is not None:
         if kind == "ieeg":
+            # get coordinate frame from digMontage
+            digpoint = raw.info['dig'][0]
+            if any(digpoint['coord_frame'] != _digpoint['coord_frame']
+                   for _digpoint in raw.info['dig']):
+                warn("Defaulting coordinate frame of iEEG coords to "
+                     "'mri' since not all digpoints "
+                     "have the same coordinate frame.")
+                coord_frame = "mri"  # defaults to MRI coordinates
+            else:
+                # get the accepted mne-python coordinate frames
+                coord_frame_int = int(digpoint['coord_frame'])
+                coord_frame = _verbose_ieeg_coordinate_frames.\
+                    get(coord_frame_int, None)
+
+                # default coordinate frame to mri if not available
+                if coord_frame is None:
+                    coord_frame = 'mri'
+
             coords = _extract_landmarks(raw.info['dig'])
 
-            # Rescale to MNE-Python "head" coord system, which is the
-            # "ElektaNeuromag" system (equivalent to "CapTrak" system)
+            # RPA, NAS and LPA shouldn't be set in iEEG
             if set(['RPA', 'NAS', 'LPA']) != set(list(coords.keys())):
                 # Now write the data to the elec coords and the coordsystem
                 _electrodes_tsv(raw, electrodes_fname,
                                 kind, overwrite, verbose)
                 _coordsystem_json(raw, unit, orient,
                                   coord_frame, coordsystem_fname, kind,
                                   overwrite, verbose)
-        elif kind != "meg":
-            warn('Writing of electrodes.tsv is not supported for kind "{}". '
-                 'Skipping ...'.format(kind))
+            else:  # noqa
+                logger.warning("Scalp EEG anatomical landmarks "
+                               "(NAS, LPA, RPA) "
+                               "shouldn't be set in iEEG. "
+                               "Skipping montage setting...")
+        elif kind == 'eeg':
+            # We only write EEG electrodes.tsv and coordsystem.json
+            # if we have LPA, RPA, and NAS available to rescale to a known
+            # coordinate system frame
+            coords = _extract_landmarks(raw.info['dig'])
+            if set(['RPA', 'NAS', 'LPA']) != set(list(coords.keys())):  # noqa
+                logger.warning("Writing EEG electrodes.tsv "
+                               "is only possible if "
+                               "anatomical landmarks (NAS, LPA, RPA) "
+                               "are present. Skipping ...")
```

Comment:
I don't know why this is necessary. For me EEG or iEEG requires the same logic. If you can write iEEG in MRI you should be able to do the same with EEG. You should only require fiducials to write in head coordinate system I think

Reply:
If that is the case, then I could wrap this in a check of the coord_frame for the raw.info['dig']?

@sappelhoff ?

Reply:
The issue is that in MNE we cannot be sure what kind of a digitization we get, because MNE does not save the coordinate system and units.

So the only way to be sure with what we work is to transform to head (and thus, automate without guessing) ... and for that, we need fiducials.

The alternative is to allow a user to manually input the coordsystem and units, but I prefer the solution here.